src/domain/flow-system/src/entities/shared/schedule.rs

```diff
@@ -113,52 +152,36 @@ mod tests {
     #[test]
     fn test_validate_invalid_expression() {
         // Try to pass invalid cron expression
-        let invalid_cron_expression = "invalid".to_string();
-        let err_result =
-            Schedule::validate_cron_expression(invalid_cron_expression.clone()).unwrap_err();
-
-        assert_eq!(
-            err_result.to_string(),
-            format!("Cron expression {} is invalid", &invalid_cron_expression),
-        );
-    }
-
-    #[test]
-    fn test_validate_expired_expression() {
-        // Try to pass invalid cron expression
-        let expired_cron_expression = "0 0 0 1 JAN ? 2024".to_string();
+        let source_with_invalid_parts = CronExpressionSource {
+            min: "invalid".to_string(),
+            ..Default::default()
+        };
         let err_result =
-            Schedule::validate_cron_expression(expired_cron_expression.clone()).unwrap_err();
+            Schedule::new_cron_schedule(source_with_invalid_parts.clone()).unwrap_err();
 
         assert_eq!(
             err_result.to_string(),
-            format!(
-                "Cron expression {} iteration has been exceeded",
-                &expired_cron_expression
-            ),
+            format!("Cron expression {} is invalid", source_with_invalid_parts),
         );
     }
 
     #[test]
     fn test_get_next_time_from_expression() {
-        let cron_expression: Schedule = "0 0 0 1 JAN ? *".to_string().into();
+        let cron_source = CronExpressionSource {
+            min: "0".to_string(),
+            hour: "0".to_string(),
+            day_of_month: "1".to_string(),
+            month: "JAN".to_string(),
+            day_of_week: "?".to_string(),
+        };
+        let cron_schedule = Schedule::new_cron_schedule(cron_source).unwrap();
 
         let current_year = Utc::now().year();
         let expected_time = Utc
             .with_ymd_and_hms(current_year + 1, 1, 1, 0, 0, 0)
             .unwrap();
 
-        let cron_time = cron_expression.next_activation_time(Utc::now()).unwrap();
-
+        let cron_time = cron_schedule.next_activation_time(Utc::now()).unwrap();
         assert_eq!(cron_time, expected_time);
     }
-
-    // Should return None if cron expression has no more iteration
```

Comment:
I just realized. Do we have any logic to remove flow if cron expression has expired?

Reply:
We don't  allow years now. How can it expire?